app/models/mixins/miq_web_server_worker_mixin.rb

```diff
@@ -121,7 +121,7 @@ def rails_server_options
       :server      => self.class.rails_server
     }
 
-    params[:Port] = port.kind_of?(Numeric) ? port : 3000
+    params[:Port] = port.kind_of?(Numeric) ? port : ENV["PORT"] || 3000
```

Comment:
Do we need to do a `Numeric` check on `ENV["PORT"]` here as well?  Might be over engineering, just thought of it because we have the Numeric check at the beginning of this ternary (probably could just be a `nil` check in all honesty, but not sure).